bottlerocket/agents/src/bin/eks-resource-agent/eks_provider.rs

```diff
@@ -194,53 +288,35 @@ impl ClusterConfig {
     }
 
     /// Create a cluster with the given config.
+    #[allow(clippy::let_and_return)]
     pub fn create_cluster(&self) -> ProviderResult<()> {
-        let status = match self {
+        let cluster_config_path = match self {
             Self::Args {
                 cluster_name,
                 region,
                 version,
-                zones,
+                zones: _,
             } => {
                 let version_arg = version
                     .as_ref()
                     .map(|version| version.major_minor_without_v())
                     .unwrap_or_else(|| DEFAULT_VERSION.to_string());
-                trace!("Calling eksctl create cluster");
-                let status = Command::new("eksctl")
-                    .args([
-                        "create",
-                        "cluster",
-                        "-r",
-                        region,
-                        "--zones",
-                        &zones.clone().unwrap_or_default().join(","),
-                        "--version",
-                        &version_arg,
-                        "-n",
-                        cluster_name,
-                        "--nodes",
-                        "0",
-                        "--managed=false",
-                    ])
-                    .status()
-                    .context(Resources::Clear, "Failed create cluster")?;
-                trace!("eksctl create cluster has completed");
-                status
+
+                create_yaml(cluster_name, region, &version_arg)?;
+                CLUSTER_CONFIG_PATH
             }
             Self::ConfigPath {
                 cluster_name: _,
                 region: _,
-            } => {
-                trace!("Calling eksctl create cluster with config file");
-                let status = Command::new("eksctl")
-                    .args(["create", "cluster", "-f", TEST_CLUSTER_CONFIG_PATH])
-                    .status()
-                    .context(Resources::Clear, "Failed create cluster")?;
-                trace!("eksctl create cluster has completed");
-                status
-            }
+            } => TEST_CLUSTER_CONFIG_PATH,
         };
+
+        trace!("Calling eksctl create cluster with config file");
```

Comment:
nit: We probably want to indicate *which* configuration file we are calling with. This could be very confusing if its using the one you generate vs one passed in and this log line wouldn't help the user discern which one is happening.